R/lockfile-metadata.R

```diff
@@ -0,0 +1,102 @@
+#' Get / Set packrat lockfile metadata
+#'
+#' Get and set metadata in the current packrat-managed project lockfile \code{packrat.lock}
+#'
+#' Project's \code{packrat.lock} contains some metadata before packages
+#' dependencies informations. The project's lockfile is created and updated
+#' programmatically by \code{\link{snapshot}}. However it could be necessary sometimes to
+#' modify manually some of those values. For example, it could be useful to set another repository
+#' CRAN url when deploying to a offline environnement.
+#'
+#' @section available metadata :
+#'
+#' \itemize{
+#' \item \code{r_version}: R version the project depends on
+#' \item \code{repos}: Name of repos and their url recorded packages can be
+#' retrieve from. Only url is recommended to change if need. Name of repos is
+#' used in package records and must be identical
+#' }
+#'
+#' @param repos A named character vector of the form \code{c(<repoName> = "<pathToRepo>")}
+#' @param rversion A one-length character vector with suitable numeric version
+#'   string. See \code{\link[base]{package_version}}
+#' @param project  The project directory. When in packrat mode, defaults to the current project;
+#'   otherwise, defaults to the current working directory.
+#' @export
+#' @rdname lockfile-metadata
+#' @name lockfile-metadata
+#' @examples \dontrun{
+#' # changes repos url
+#' repos <- old_repos <- get_lockfile_metadata("repos")
+#' repos
+#' repos["CRAN"] <- "https://cran.r-project.org/"
+#' set_lockfile_metadata(repos = repos)
+#' get_lockfile_metadata("repos")
+#' # setting back old state
+#' # set_lockfile_metadata(repos = old_repos)
+#'
+#' # changes R version
+#' rver <- old_rver <- get_lockfile_metadata("r_version")
+#' rver
+#' rver <- "3.4.1"
+#' set_lockfile_metadata(r_version = rver)
+#' get_lockfile_metadata("r_version")
+#' # Setting back old state
+#' # set_lockfile_metadata(r_version = old_rver)
+#' }
+set_lockfile_metadata <- function(repos = NULL, r_version = NULL, project = NULL) {
```

Comment:
Most of the Packrat APIs take the project as the first argument -- can we follow that convention here?

Reply:
I followed the convention of `set_opts` and `get_opts` that I find close in functionality. `project` is last in `get_opts` and second in `set_opts`. From usage :
```r
get_opts(options = NULL, simplify = TRUE, project = NULL)
set_opts(..., project = NULL, persist = TRUE)
```
bad idea ? 
Still want me to change ?

Reply:
Good point! I think you're right; let's leave it as is then.